core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+        #   ({{'%0.6f' | format(answer[0]|float)}}, {{'%0.6f'|format(answer[1]|float)}})
+        supported_rule_types = ['Within', 'NotWithin']
+        if rule_str != cls._DEFAULT_RULESPEC_STR:
+            if rule_spec.rule_type not in supported_rule_types:
+                return (
+                    None,
+                    'Unsupported rule type encountered while attempting to '
+                    'reconstitute CoordTwoDim object: %s' % rule_spec.rule_type)
+
+        # Match the pattern: '(real, real)' to extract the coordinates.
+        pattern = re.compile(
+            r'\((?P<x>-?\d+\.?\d*), (?P<y>-?\d+\.?\d*)\)')
+        match = pattern.match(answer_str)
+        if not match:
+            # Prior to #380ea2 on 03 June 2014, the answers were stored as
+            # stringifications of the CoordTwoDim object using schema_utils, so
+            # the pattern for reconstruction is slightly different.
+            pattern = re.compile(
+                r'\[(?P<x>-?\d+\.?\d*), (?P<y>-?\d+\.?\d*)\]')
+            match = pattern.match(answer_str)
+            if not match:
+                # Prior to #d60ccb on 16 March 2014, the answers were stored as
+                # strings based on how they were represented in the frontend, as
+                # this commit predates the creation of schema_utils. The pattern
+                # is different than the other two cases.
+                pattern = re.compile(
+                    r'(?P<x>-?\d+\.?\d*),(?P<y>-?\d+\.?\d*)')
+                match = pattern.match(answer_str)
+                if not match:
+                    return (
+                        None, 'Bad answer string in InteractiveMap rule.')
+        coord_two_dim_list = [
+            float(match.group('x')), float(match.group('y'))
+        ]
+        return cls._normalize_raw_answer_object(
+            objects.CoordTwoDim, coord_two_dim_list, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_item_selection_input(
+            cls, state, answer_group_index, rule_spec, rule_str, answer_str):
+        # The Jinja representation for SetOfHtmlString answer strings is:
+        #   {{ answer }}
+        supported_rule_types = [
+            'Equals', 'ContainsAtLeastOneOf', 'DoesNotContainAtLeastOneOf'
+        ]
+        if rule_str != cls._DEFAULT_RULESPEC_STR and (
+                rule_spec.rule_type not in supported_rule_types):
+            return (
+                None,
+                'Cannot reconstitute ItemSelectionInput object without an '
+                'Equals rule.')
+        option_list = ast.literal_eval(answer_str)
+        if not isinstance(option_list, list):
+            return (
+                None,
+                'Bad answer string in ItemSelectionInput Equals rule.')
+        return cls._normalize_raw_answer_object(
+            objects.SetOfHtmlString, option_list, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_logic_proof(
```

Comment:
I don't actually see any real occurrences of CheckedProof in the final data, which may be because they're being stripped by the purge job. This was definitely one of the harder answers to migrate, and we may still have default answers that are using this, but I suspect it's so low in usage that even if there's a mistake here it won't be too bad (since it isn't causing any runtime issues with the migration).